core/src/cluster_info_vote_listener.rs

```diff
@@ -459,7 +465,9 @@ impl ClusterInfoVoteListener {
             );
             match confirmed_slots {
                 Ok(confirmed_slots) => {
-                    confirmation_verifier.add_new_optimistic_confirmed_slots(confirmed_slots);
+                    confirmation_verifier
+                        .add_new_optimistic_confirmed_slots(confirmed_slots.clone());
+                    let _ = cluster_confirmed_slot_sender.send(confirmed_slots);
```

Comment:
@behzadnouri I think this can also be expanded to check for vote hashes for duplicate slots once you've added those hashes to gossip as well